### ISSUE TITLE
Query bean miss field definition (when using javax.validation 2.0.1.Final)

### ISSUE THREAD

**Issue by 54446776** [ebean-orm/querybean-generator#18]
If an entity field with a size annotation, it will be missed in generated query bean:
@Column(unique = true, columnDefinition = "text")
@Size(min = 1, max = 100)
String name;

The name field will be not exist in query bean.

Root cause is the field type returned by field.asType is: 
(@javax.validation.constraints.Size(min=1, max=10) :: java.lang.Strring)

It should be escaped by regexp in ProcessingContext.getPropertyType method:
private static final Pattern ESCAPER = Pattern.compile("\\(.+ :: (.+)\\)");
private static String excape(String typeNameWithAnno) {
    Matcher m = ESCAPER.matcher(typeNameWithAnno);
    if (m.matches()) {
      return m.group(1);
    } else {
      return typeNameWithAnno;
    }
 }
...
PropertyType type = propertyTypeMap.getType(excape(currentType.toString()));

**Comment by rbygrave** [comment]
I can not reproduce this issue ... I tried using https://github.com/ebean-orm-examples/example-java8

Are you able to fork that and create a branch that fails?

**Comment by 54446776** [comment]
I found the root cause:

validation-api 1.0.0 is fine:
<dependency>
      <groupId>javax.validation</groupId>
      <artifactId>validation-api</artifactId>
      <version>1.0.0.GA</version>
</dependency>

But validation-api 2.0.1 fail:
<dependency>
      <groupId>javax.validation</groupId>
      <artifactId>validation-api</artifactId>
      <version>2.0.1.Final</version>
</dependency>

**Comment by rbygrave** [comment]
Ah cool, thanks.  I'll have a look at reproducing with that later ...

Cheers, Rob.

**Comment by 54446776** [comment]
Hi rbygrave,

Could you please reproduce this issue and fix it? 
[Hibernate Validator 5x is not compatible with validation-api 1.0.x](https://stackoverflow.com/questions/24652753/java-lang-noclassdeffounderror-javax-validation-parameternameprovider), and I have to use validation-api 2.x.

**Comment by rbygrave** [comment]
Ok yes, reproduced with

```xml
    <dependency>
      <groupId>javax.validation</groupId>
      <artifactId>validation-api</artifactId>
      <version>2.0.1.Final</version>
      <!--<version>1.0.0.GA</version>-->
    </dependency>
```
Working on it ...

**Comment by rbygrave** [comment]
Thanks for the investigation etc.  Fixed and released as 11.1.2
